theseus/core/loss.py

```diff
@@ -23,20 +25,20 @@ def linearize(x: torch.Tensor, raidus: torch.Tensor) -> torch.Tensor:
 class WelschLoss(Loss):
     @staticmethod
     def evaluate(x: torch.Tensor, radius: torch.Tensor) -> torch.Tensor:
-        return radius - radius * torch.exp(-x / radius)
+        return radius - radius * torch.exp(-x / (radius + _LOSS_EPS))
```

Comment:
I added some EPS for potentially problematic operations, @fantaosha can double check if these make sense. We should later consolidate these as part of #70.

Reply:
I would like to replace `radius` with `log_radius` since there are no guarantees that `radius` is nonnegative. Instead, `radius = torch.exp(log_radius)` always yields a positive one.